src/track/track.js

```diff
@@ -67,67 +68,71 @@ export default class Track extends Component {
   }
 
   eventListeners = []
+  isScrolling = false
+  shouldSelfCorrect = () =>
+    !this.props.preventSnapping &&
+    !this.state.isAnimating &&
+    !this.isScrolling &&
+    !this.isInteracting()
 
   componentDidMount () {
     this.DOMNode = findDOMNode(this.track)
+    this.isInteracting = hasOngoingInteraction(this.DOMNode)
 
     // These are not a part of component state since we don't want
     // incure the overhead of calling setState. They are either cached
     // values or state only the onScrollEnd callback cares about and
     // are not important to the rendering of the component.
     this.childCount = this.track.children.length
-    let isScrolling = false
-    const isInteracting = hasOngoingInteraction(this.DOMNode)
-    const shouldSelfCorrect = () =>
-      !this.props.preventSnapping &&
-      !this.state.isAnimating &&
-      !isScrolling &&
-      !isInteracting()
 
     const slideBy = {
-      left: -this.state.slideBy,
-      right: this.state.slideBy
+      left: () => -this.state.slideBy,
+      right: () => this.state.slideBy
     }
 
     this.eventListeners = [
       ...this.eventListeners,
 
-      onScrollStart(() => { isScrolling = true }),
+      onScrollStart(() => { this.isScrolling = true }),
 
-      on('touchstart')(() => { isScrolling = true })(this.track),
+      on('touchstart')(() => { this.isScrolling = true })(this.track),
 
       onScrollEnd(() => {
-        isScrolling = false
-        if (shouldSelfCorrect()) {
+        this.isScrolling = false
+        if (this.shouldSelfCorrect()) {
           this.setState({ isAnimating: true })
           this.slideTo(this.getNearestSlideIndex()).catch(noop)
         }
         this.setState({ isAnimating: false })
       }, { target: this.DOMNode }),
 
       on('touchend')(() => {
-        if (shouldSelfCorrect()) {
+        if (this.shouldSelfCorrect()) {
           this.setState({ isAnimating: true })
           this.slideTo(this.getNearestSlideIndex()).catch(noop)
         }
       })(this.track),
 
       onSwipe((direction) => {
-        this.slideTo(this.state.activeIndex + (slideBy[direction] || 0)).catch(noop)
+        this.slideTo(this.state.activeIndex + (slideBy[direction]() || 0)).catch(noop)
       })(this.track)
 
     ]
 
     this.slideTo(this.props.startAt, { immediate: true }).catch(noop)
   }
 
-  componentWillUnmount () {
-    this.eventListeners.map((fn) => fn())
+  componentWillUnmount () { this.eventListeners.map((fn) => fn()) }
```

Comment:
`forEach` here because you are performing side effects and not using the new array `map` is creating.